source/kraken/maintenance_worker.cpp

```diff
@@ -56,14 +56,116 @@ namespace bg = boost::gregorian;
 
 namespace navitia {
 
+void MaintenanceWorker::run() {
+    // try to load data from disk
+    load_data();
+
+    if (!is_data_loaded()) {
+        // The data is not loaded, so we wait for a reload message on the task queue
+        // that results in a successful data load.
+        // We may be disconnected from rabbitmq during this wait,
+        // and reconnection to rabbitmq is handled by this while(true) loop
+        while (true) {
+            try {
+                open_channel_to_rabbitmq();
+                create_task_queue();
+                // will return when a reload message on the task queue
+                // arrived and resulted in a successful data load
+                listen_to_task_queue_until_data_loaded();
+                // data is loaded, let's break from the while(true)  loop
+                break;
+            } catch (const std::runtime_error& ex) {
+                LOG4CPLUS_ERROR(logger, "Connection to rabbitmq failed: " << ex.what());
+                data_manager.get_data()->is_connected_to_rabbitmq = false;
+                channel_opened = false;
+                std::this_thread::sleep_for(std::chrono::seconds(1));
+            }
+        }
+    }
+
+    // if rabbitmq was not connected, we try to connect to rabbitmq
+    // When connection is successfull, we don't want to load_data(), we just listen to the queues
+    while (true) {
+        try {
+            // will do nothing is channel is already opened
+            open_channel_to_rabbitmq();
+            // will do nothing if the queue already exists
+            create_task_queue();
+
+            create_realtime_queue();
+            listen_rabbitmq();
+        } catch (const std::runtime_error& ex) {
+            LOG4CPLUS_ERROR(logger, "Connection to rabbitmq failed: " << ex.what());
+            data_manager.get_data()->is_connected_to_rabbitmq = false;
+            channel_opened = false;
+            std::this_thread::sleep_for(std::chrono::seconds(1));
```

Comment:
```suggestion
            std::this_thread::sleep_for(std::chrono::seconds(conf.broker_sleeptime()));
```

Reply:
I used `BROKER.reconnect_wait` as the new parameter